pkg/vsphere/tasks/waiter.go

```diff
@@ -41,21 +46,21 @@ type Task interface {
 
 // Wait wraps govmomi operations and wait the operation to complete
 // Sample usage:
-//    info, err := Wait(ctx, func(ctx) (*TaskInfo, error) {
-//       return vm.Reconfigure(ctx, config)
+//    info, err := Wait(ctx, func(ctx), sess *session.Session, *object.Reference, (*object.Reference, *TaskInfo, error) {
+//       return vm, vm.Reconfigure(ctx, sess, target, config)
 //    })
-func Wait(ctx context.Context, f func(context.Context) (Task, error)) error {
-	_, err := WaitForResult(ctx, f)
+func Wait(ctx context.Context, sess *session.Session, target object.Reference, f func(context.Context) (Task, error)) error {
+	_, err := WaitForResult(ctx, sess, target, f)
 	return err
 }
 
 // WaitForResult wraps govmomi operations and wait the operation to complete.
 // Return the operation result
 // Sample usage:
-//    info, err := WaitForResult(ctx, func(ctx) (*TaskInfo, error) {
-//       return vm.Reconfigure(ctx, config)
+//    info, err := WaitForResult(ctx, sess *session.Session, *object.Reference, func(ctx) (*TaskInfo, error) {
+//       return vm, vm.Reconfigure(ctx, config)
 //    })
-func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
+func WaitForResult(ctx context.Context, sess *session.Session, target object.Reference, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
```

Comment:
The change itself turns to be much more confusing than everybody thought. But the result is that we have to do it in tasks.WaitForResult, cause every VM operation might end up with this failure, and the solution is not only to fix the error, but also retry task.
If we move the fix out of WaitForResult, we'll need to modify every VM operation call, patch with this workaround, that's not what we should do.